src/projects/shared/site/src/containers/Header/Right/OpenFileButton.tsx

```diff
@@ -19,7 +19,7 @@ export const OpenFileButton = () => {
     const fileInput = React.useRef<HTMLInputElement>(null);
 
     const load = (file: File) => {
-        if (isSaved || window.confirm(OVERWRITE_CIRCUIT_MESSAGE))
+        if (isSaved && window.confirm(OVERWRITE_CIRCUIT_MESSAGE))
```

Comment:
? The logic is supposed to be "if the circuit is already saved" OR they confirm overwriting it, then load it. If it's an AND, then they have to save their circuit and confirm overwriting it; if it's not working it's probably an issue with isSaved not being set correctly?